src/back/spv/instructions.rs

```diff
@@ -219,67 +219,26 @@ impl super::Instruction {
         instruction
     }
 
+    #[allow(clippy::too_many_arguments)]
     pub(super) fn type_image(
         id: Word,
         sampled_type_id: Word,
         dim: spirv::Dim,
+        depth: bool,
```

Comment:
This is internal API, so it's not that important, but generally we shouldn't accept booleans.
Creating a bitflags type for these would be nice.

Reply:
I did this, it was a great cleanup. Noted for the future.